### ISSUE TITLE
unexpected behavior of use_debounce

### ISSUE THREAD

**Issue by jgraydus** [jetli/yew-hooks#39]
It seems that `use_debounce` returns a callback in a triggered state. That is, after the given time elapses the original callback is invoked whether or not the debounced version was ever invoked. This is not the behavior that I would expect. Is this the desired behavior for this hook?

The following code demonstrates:

```rust
use gloo_console::log;
use yew::prelude::*;
use yew_hooks::use_debounce;

#[function_component(App)]
fn app() -> Html {
  let f = use_debounce(|| {
    log!("this should never print, but it does");
  }, 1000);

  html! {}
}

fn main() {
  yew::Renderer::<App>::new().render();
}
```

**Comment by tgrushka** [comment]
Yes, this is definitely a bug, because `use_debounce` calls `use_timeout` directly, which runs its callback unconditionally.

But there is an additional fundamental bug in `use_debounce`, `use_debounce_state`, and `use_timeout`. If you ever try to call `f.run()`, you'll also note it never gets called again after the initial call!

This is due to:
1. the callback being taken out of the `use_ref_mut` and never put back;
2. the use of `FnOnce()` instead of `Fn()` in `use_timeout` and the others, which results in the callback only ever being called once:

From `FnOnce` docs from `rustlib/src/rust/library/core/src/ops/function.rs`:

```rust
/// Instances of `FnOnce` can be called, but might not be callable multiple
/// times. Because of this, if the only thing known about a type is that it
/// implements `FnOnce`, it can only be called once.
```

I've managed to fix these issues and will do a PR.
